src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java

```diff
@@ -0,0 +1,191 @@
+package com.denizenscript.clientizen.events;
+
+import com.denizenscript.denizencore.events.ScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class KeyPressReleaseEvent extends ScriptEvent {
+
+	public static KeyPressReleaseEvent instance;
+
+	public KeyboardKeys key;
+	public boolean pressed;
+
+
+	public KeyPressReleaseEvent() {
+		registerCouldMatcher("keyboard <'key'> pressed|released");
+		instance = this;
+	}
+
+	@Override
+	public boolean matches(ScriptPath path) {
+		String keyMatcher = path.eventArgLowerAt(1);
+		if (!keyMatcher.equals("key") && !runGenericCheck(keyMatcher, key.getName())) {
+			return false;
+		}
+		if (pressed != path.eventArgLowerAt(2).equals("pressed")) {
```

Comment:
That check feels a bit iffy as opposed to the `thing && !arg matches thing` style checks that are used in other places, might want to change that for consistency but ig it's fine either way

Reply:
That would not account for the released event. If the arg is `pressed` and the key was released, the boolean `pressed` will be false, missing out on that check entirely and letting it pass.

Reply:
Ah right, my bad - disregard this